src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -839,10 +855,11 @@ private static string GetSettingsFileNameFromDir(string directory)
 
         private static IEnumerable<string> GetSettingsFilePaths(string root)
```

Comment:
Why not have this method accept nullable?

```suggestion
        private static IEnumerable<string> GetSettingsFilePaths(string? root)
```

Reply:
because to me it's non-sensical to call a method that finds all parent directories of a null path. Yes, it would allow us to have one less line, but we don't win anything for code golf, and I feel like correctness is nicer than fewer lines.  But that's my personal preference 😁

Reply:
True, in a perfect world this should throw an `ArgumentNullException` but that's probably not possible to change behavior.